packages/react/src/components/Forms/TextArea/index.spec.tsx

```diff
@@ -0,0 +1,193 @@
+      </div>,
+    );
+
+    const textarea: HTMLTextAreaElement = screen.getByRole("textbox", {
+      name: "Describe your job",
+    });
+    const textarea2: HTMLTextAreaElement = screen.getByRole("textbox", {
+      name: "Describe your hobbies",
+    });
+    const label = screen.getByText("Describe your job")!.parentElement!;
+    expect(Array.from(label.classList)).toContain("placeholder");
+
+    // Clicking on the textarea should remove the placeholder class.
+    await user.click(textarea);
+
+    expect(Array.from(label.classList)).not.toContain("placeholder");
+
+    // Writing something should remove the placeholder class too.
+    await user.type(textarea, "Developer");
+
+    expect(Array.from(label.classList)).not.toContain("placeholder");
+
+    // Clearing the textarea and focus out should add the placeholder class
+    await user.clear(textarea);
+    await user.click(textarea2);
+
+    expect(Array.from(label.classList)).toContain("placeholder");
+  });
+
+  it("renders with state=success", async () => {
+    render(<TextArea label="First name" state="success" />);
+    expect(document.querySelector(".c__field--success")).toBeInTheDocument();
+    expect(document.querySelector(".c__field--success")).toBeInTheDocument();
+  });
+
+  it("renders with state=error", async () => {
+    render(<TextArea label="First name" state="error" />);
+    expect(document.querySelector(".c__field--error")).toBeInTheDocument();
+    expect(document.querySelector(".c__field--error")).toBeInTheDocument();
+  });
+
+  it("renders disabled", async () => {
+    const user = userEvent.setup();
+    render(<TextArea label="Describe your job" disabled={true} />);
+    const textarea: HTMLTextAreaElement = screen.getByRole("textbox", {
+      name: "Describe your job",
+    });
+    expect(
+      document.querySelector(".c__textarea__wrapper--disabled"),
+    ).toBeInTheDocument();
+    expect(textarea.value).toEqual("");
+    // Disabled textareas should not be able to type.
+    await user.type(textarea, "Developer");
+    expect(textarea.value).toEqual("");
+  });
+
+  it("renders with text", async () => {
+    render(<TextArea label="Describe your job" text="Some text" />);
+    screen.getByText("Some text");
+  });
+
+  it("renders with text items", async () => {
+    render(
+      <TextArea
+        label="Describe your job"
+        textItems={[
+          "Text too long",
+          "Wrong choice",
+          "Must contain at least 9 characters, uppercase and digits",
+        ]}
+      />,
+    );
+    expect(
+      screen.getAllByRole("listitem").map((item) => item.textContent),
+    ).toEqual([
+      "Text too long",
+      "Wrong choice",
+      "Must contain at least 9 characters, uppercase and digits",
+    ]);
+  });
+
+  it("renders with text and text right", async () => {
+    render(<TextArea label="Describe your job" rightText="Some text right" />);
+    screen.getByText("Some text right");
```

Comment:
Should we check that the element retrieved has the class `c__field__text__right` ?

Reply:
yep!